src/test/java/utils/JavaFXThreadingRule.java

```diff
@@ -0,0 +1,99 @@
+package utils;
+
+import javafx.application.Platform;
+import javafx.embed.swing.JFXPanel;
+import org.junit.Rule;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import javax.swing.*;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * A JUnit {@link Rule} for running tests on the JavaFX thread and performing
+ * JavaFX initialisation.  To include in your test case, add the following code:
+ *
+ * <pre>
+ * {@literal @}Rule
+ * public JavaFXThreadingRule jfxRule = new JavaFXThreadingRule();
+ * </pre>
+ *
+ * @author Andy Till
+ *
+ */
+public class JavaFXThreadingRule implements TestRule {
+
+    /**
+     * Flag for setting up the JavaFX, we only need to do this once for all tests.
+     */
+    private static boolean jfxIsSetup;
+
+    @Override
+    public Statement apply(Statement statement, Description description) {
+
+        return new OnJFXThreadStatement(statement);
+    }
+
+    private static class OnJFXThreadStatement extends Statement {
+
+        private final Statement statement;
+
+        private Throwable rethrownException = null;
+
+        public OnJFXThreadStatement(Statement aStatement) {
+            statement = aStatement;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
+
+            if(!jfxIsSetup) {
+                setupJavaFX();
+
+                jfxIsSetup = true;
+            }
+
+            final CountDownLatch countDownLatch = new CountDownLatch(1);
+
+            Platform.runLater(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        statement.evaluate();
+                    } catch (Throwable e) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [A catch statement should never catch throwable since it includes errors.](https://www.codacy.com/app/POPBL6/front-end/file/2720643984/issues/source?bid=3316499&fileBranchId=3347636#l64)